source/elements/oneDAL/source/common_interface/error_handling.rst

```diff
@@ -1,3 +1,127 @@
+.. _error_handling:
+
 ==============
 Error handling
 ==============
+
+oneDAL error handling relies on mechanism of C++ exceptions. If an error occurs,
+it must be propagated at the point of a function call where it can be caught
+using С++ exception handlers.
+
+Exceptions classification
+=========================
+
+Classification is aligned with C++ STL classification. oneDAL base exception
+classes are abstract. Leaf classes are also inherited from the corresponding STL
+exceptions. Only these classes can be thrown.
+
+The root of exception hierarchy is ``onedal::exception``. All throwing
+exceptions are inheritors of this one.
+
+::
+
+    class onedal::exception;
+
+.. list-table::
+   :widths: 30 65 5
+   :header-rows: 1
+
+   * - Exception
+     - Description
+     - Abstract
+   * - onedal::exception
+     - Base class for all throwing exceptions.
+     - Yes
+
+
+``onedal::logic_error``, ``onedal::runtime_error`` and ``onedal::bad_alloc`` are inheritors of ``onedal::exception``.
+
+::
+
+    class onedal::logic_error : public onedal::exception;
+    class onedal::runtime_error : public onedal::exception;
+    class onedal::bad_alloc :  public onedal::exception, public std::bad_alloc;
+
+.. list-table::
+   :widths: 30 65 5
+   :header-rows: 1
+
+   * - Exception
+     - Description
+     - Abstract
+   * - onedal::logic_error
+     - Reports violations of preconditions and invariants.
+     - Yes
+   * - onedal::runtime_error
+     - Reports violations of postconditions and other errors that are due to events beyond the scope of the program.
+     - Yes
+   * - onedal::bad_alloc
+     - Reports failure to allocate storage.
+     - No
+
+Precondition and invariant errors that are represented by ``onedal::logic_error``
+divide in subgroups.
+
+::
+
+    class onedal::invalid_argument :  public onedal::logic_error, public std::invalid_argument;
```

Comment:
Introduce the description similar to the one I mentioned earlier but using a short sentence, w/o listing all those exceptions